storage/src/archive/prunable/storage.rs

```diff
@@ -72,7 +72,7 @@ pub struct Archive<T: Translator, E: Storage + Metrics, K: Array, V: Codec> {
     // to its corresponding index. To avoid iterating over this keys map during pruning, we map said
     // indexes to their locations in the journal.
     keys: Index<T, u64>,
-    indices: BTreeMap<u64, Location>,
+    indices: BTreeMap<u64, Vec<(K, Location)>>,
```

Comment:
A big goal of `archive::prunable` is to avoid storing the keys in memory (if we were open to this, we could just remove the `Index`).

I think the right approach is to just store a `Vec<Location>` and iterate over all locations when serving.